lib/backend/k8s/resources/retrywrapper.go

```diff
@@ -0,0 +1,163 @@
+// Copyright (c) 2017 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+
+	log "github.com/Sirupsen/logrus"
+)
+
+const (
+	maxActionRetries = 5
+)
+
+// retryWrapper implements the K8sResourceClient interface and is used to wrap
+// another K8sResourceClient to provide retry functionality when the failure
+// case is of type retryError.
+type retryWrapper struct {
+	client K8sResourceClient
+}
+
+// retryError is an error type used to indicate to the retryWrapper to retry
+// a specific action.
+//
+// If the action is retried the max number of times, then the retryWrapper will
+// return the underlying error.
+type retryError struct {
+	err error
+}
+
+func (r retryError) Error() string {
+	return r.err.Error()
+}
+
+func (r *retryWrapper) Create(object *model.KVPair) (*model.KVPair, error) {
+	var kvp *model.KVPair
+	var err error
+	for i := 0; i < maxActionRetries; i++ {
+		if kvp, err = r.client.Create(object); err == nil {
+			// No error, exit returning the KVPair.
+			return kvp, nil
+		} else {
+			return nil, err
```

Comment:
Something's not clicking here for me... isn't this just going to return after one retry whether or not the Create succeeds / fails?